scripts/Calc_hru_wgtavg_nc.py

```diff
@@ -89,115 +80,62 @@ def getNetCDFAtt(fn, varname,attName):
-  ncfile.source = os.path.dirname(os.path.abspath(__file__))+__file__[1:]
-  
-  ncfile.close()
-
-def compAvgVal(nc_wgt,nc_in,varname):
+def compAvgVal(nc_wgt, nc_in, varname, chunk):
   """Compute areal weighted avg value of <varname> in <nc_in> for each hru based on hru's weight in <nc_wgt>""" 
-  wgt = wgtnc(nc_wgt)             #instantaneous of wgtnc object
-  hruIdName = wgt.getHruIdName()  # Get precise name of variable for hru id 
+  wgt       = wgtnc(nc_wgt)       #instantaneous of wgtnc object
   hruIDs    = wgt.getHruID()      #get hruID list 
+  hrudata   = split(hruIDs,NPROC)
+  hrulist   = hrudata[chunk]
 
   dataVal  = getNetCDFData(nc_in,varname) # Get data value 
   FillVal  = getNetCDFAtt(nc_in,varname,'_FillValue') # Get data value 
-  IdVal    = getNetCDFData(nc_in,IDNM) 
+  ghruid   = getNetCDFData(nc_in,HDIM)
   dim1size = dataVal.shape[0]
-
   #Initialize wgtVal[ntime,nhru] 
-  wgtVal = np.zeros((dim1size,len(hruIDs)))
+  wgtVal = np.ones((dim1size,len(hrulist)))*-9999.0
   #Loop through each hru polygon 
-  for i in range(len(hruIDs)):
-    print "Computing weighted values over hru%d" %hruIDs[i]
-    # Get list of wgt, lat, and lon for corresponding hru
-    (wgtval, overlapsId, lat, lon, overlaps)=wgt.getWgtHru(hruIDs[i])
-    wgtArray = np.asarray(wgtval)
-    
+  for i in range(len(hrulist)):
+    print hrulist[i]
+    # Get list of wgt for corresponding hru
+    (wgtArray, overlapsId, overlaps)=wgt.getWgtHru(hrulist[i])
     #Count missing cells (No values in original input polygon) and valid cells
     numvoid = 0
     # Go through wgt list and replace value with zero for following cases
     # where overlapping polygon has missing value - case1
     # where overlapping polygon is outside nc_wgt domain - case2
-    for j in range(len(wgtArray)): # Go through each overlapping polygon
-
-      if overlapsId[j] >= 0:              # if there is at least one overlapping polygon
-        # find index of grid cell that match up with hru id of overlapsId[j] 
-        ij=np.where(IdVal==overlapsId[j])
-        # if nc_in netCDF does not cover nc_wgt domain - case2  
-        if len(ij[0]) == 0:
-          numvoid = numvoid+1
-          wgtArray[j]=0
-        else:
-          a = dataVal[10,ij[0][0],ij[1][0]].tolist()
-          #if value of overlapping polygon is missing data -case1
-          if a == None: 
-            numvoid = numvoid+1
-            wgtArray[j] = 0 
-    
-    #Initialize Val2[ntime] 
-    Val2 = np.zeros((dim1size))
+    a = np.zeros( (dim1size,len(wgtArray)) )
+    for j in range(overlaps):                     # Go through each overlapping polygon
+      # find index of grid cell that match up with hru id of overlapsId[j]
+      ij=np.column_stack(np.where(ghruid==overlapsId[j]))
+      # if nc_in netCDF does not cover nc_wgt domain - case2
+      if len(ij[0]) == 0:
+        numvoid+=1
+        wgtArray[j]=0.0
+      else:
+        a[:,j] = dataVal[:,ij[0,0],ij[0,1]]
```

Comment:
This is some ugly indexing. What is going on here?

Reply:
yes, ugly.   `dataVal` is [time, y, x] 3D runoff grid, and wanted to get indices for x and y. 
`ghruid`is [y,x] array including overlap_ids and get the indices from `ij=np.column_stack(np.where(ghruid==overlap_ids[j]))`   is there any better way (maybe rewrite in xarray)?